src/ReadOnlyArrays.jl

```diff
@@ -38,6 +38,12 @@ struct ReadOnlyArray{T,N,P} <: AbstractArray{T,N}
     parent::P
     ReadOnlyArray(parent::AbstractArray{T,N}) where{T,N} =
         new{T, N, typeof(parent)}(parent)
+    ReadOnlyArray{T}(parent::AbstractArray{T,N}) where{T,N} =
+        new{T, N, typeof(parent)}(parent)
+    ReadOnlyArray{T, N}(parent::AbstractArray{T,N}) where {T, N} =
+        new{T, N, typeof(parent)}(parent)
+    ReadOnlyArray{T, N, P}(parent::P) where {T, N, P<:AbstractArray{T, N}} =
```

Comment:
I would keep only one inner constructor and define the other variants as outer constructors.

Also could you please make sure to consistently use whitespace after comma (the same comment applies to code below).

Reply:
Fixed